### PR TITLE
move nvs memory generation script here

### PR DESCRIPTION
**Pull request by sven-hm**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,137 @@
...
 
 - [ubirch-esp32-storage](https://github.com/ubirch/ubirch-esp32-storage)
 - [ubirch-protocol](https://github.com/ubirch/ubirch-protocol.git)
+
+# ubirch ID handling and registration
+
+In order to simplify the management of ubirch identities and the correspondet keys and configurations, 
+the built-in partition `nvs` of the flash storage is used ([Built-in Partition Tables](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/partition-tables.html#built-in-partition-tables))
+where this data is stored, without the need of modifying the codebase.
+The application is then able to load and store the values in the `nvs` via specific functions from [ubirch-esp32-storage](https://github.com/ubirch/ubirch-esp32-storage)
```
Comment (sven-hm) [inline README.md:31]: This reference points to the repository itself. Is this intended?

File: README.md [hunk 2]
```diff
@@ -1,23 +1,137 @@
...
+The python script has some optional input arguments, which will be explained by this examplary usage:
+```[bash]
+$ python create_nvs_memory.py --json device_config.json --stage demo --token <insert your token string here> --out out.csv 
+```
+- `--json device_config.json` is necessary for identity configurations to be stored. 
+If no `--out` argument is specified, the output file will be `device_config.csv`
+- `--stage demo` specifies the stage of the ubirch backend, which is needed to provide the public key. 
+If this argument is not used, the default value `prod` will be used.
+- `--token <insert your token string here>` specifies the JWT token.
+- `--out out.csv` specifies the output file.
```
Comment (sven-hm) [inline README.md]: I would also mention:
- if no `--json` is given, no devices will be pre-configured (e.g. as in the gateway example)
- `--token` is only needed if you want to use automatic device registration

File: README.md [hunk 3]
```diff
@@ -1,23 +1,137 @@
...
+$ python create_nvs_memory.py --stage demo --out gateway_memory.csv --token <insert your token string here>
+```
+
+## Create binary-file for NVS
+
+Now you have a csv-file and can continue to create the nvs-flash-partition description binary-file (for more details compare
+[the esp-idf documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/storage/nvs_partition_gen.html)) by running:
+```bash
+$ python $IDF_PATH/components/nvs_flash/nvs_partition_generator/nvs_partition_gen.py generate my_device_config.csv my_device_config.bin 0x3000
+```
```
Comment (sven-hm) [inline README.md:119]: Maybe we should note that the python version provided by esp-idf should be used to run the `nvs_partition_gen.py` script (i think it requires python3 and some python-modules that the user might not have installed in his python environment) and can also be used for our `create_nvs_memory.py` script.
The esp-idf's python is activated when running the `.../esp-idf/export.sh` script.

File: token_handling.h [hunk 4]
```diff
@@ -36,7 +36,10 @@ extern "C" {
 typedef uint8_t ubirch_token_state_t;
 
 /*!
- * Check token state
+ * Check token state.
+ *
+ * @param state_bit_mask The bit mask to check against
+ * @return true if the state of /p state_bit_mask coresponds with stae of token
```
Comment (sven-hm) [inline token_handling.h]: typo: stae -> state

File: README.md (hunk 1 above)
Comment (gruenwaldi) [reply README.md:31]: no it is not. the reference goes to `esp32-storage`, not `esp32-key-storage`

File: README.md (hunk 3 above)
Comment (gruenwaldi) [reply README.md:119]: I would say, this is up to the users to figure this out.

**Pull request merged**